pandas/tests/io/parser/test_network.py

```diff
@@ -15,42 +15,36 @@
 from pandas.io.parsers import read_csv, read_table
 
 
-class TestCompressedUrl(object):
+@pytest.fixture(scope='module')
+def salaries_table():
+    path = os.path.join(tm.get_data_path(), 'salaries.csv')
+    return read_table(path)
 
-    compression_to_extension = {
-        'gzip': '.gz',
-        'bz2': '.bz2',
-        'zip': '.zip',
-        'xz': '.xz',
-    }
 
-    def setup(self):
-        path = os.path.join(tm.get_data_path(), 'salaries.csv')
-        self.local_table = read_table(path)
-        self.base_url = ('https://github.com/pandas-dev/pandas/raw/master/'
-                         'pandas/io/tests/parser/data/salaries.csv')
+@pytest.mark.parametrize(
+    "compression,extension", [('gzip', '.gz'), ('bz2', '.bz2'),
+                              ('zip', '.zip'), ('xz', '.xz')])
+def test_compressed_urls(salaries_table, compression, extension):
```

Comment:
IIRC, decorators need to fully copy the whole argument signature. This is not too bad on Python 3.4+ because `functools.wraps` does it already, but it's a bit of a pain on 2.7. The easiest is to use the `decorator` package, though on `matplotlib`, I just replaced the decorators like that with an automatic fixture and some markers.